### PR TITLE
update supported arches for kn-workflow-cli-artifacts image

### PR DESCRIPTION
**Pull request by rgdoliveira**
As discussed in slack channel, for kn cli we should only support amd64 and aarch64, IBM P/Z is not supported and there will be no rpm build of it.

Related PRs:
- https://github.com/kiegroup/kogito-serverless-operator/pull/36
- https://github.com/kiegroup/kogito-serverless-operator/pull/58

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by rgdoliveira** [comment]
As discussed on slack channel, the kn workflow rpm will be built for x86_64 and aarch64 only, IBM Z and P are not required.

**Review by mareknovotny — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Comment by rgdoliveira** [comment]
Thanks for approving it, I will just wait for the rpm part to be built and then I can merge this one (or change something if required)

**Review by domhanak — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

File: images/modules/com.redhat.osl.cli.packager/install.sh
Comment (rgdoliveira) [inline images/modules/com.redhat.osl.cli.packager/install.sh]: The image is failing to build in this part related to Windows and it seems it is failing when invoking the `zip` command.
 @ricardozanini could you please take a look at it? I sent in slack the command to build the image with correct URL.

Comment (ricardozanini) [reply images/modules/com.redhat.osl.cli.packager/install.sh]: What's the error?

Comment (rgdoliveira) [reply images/modules/com.redhat.osl.cli.packager/install.sh]:
```
+ mkdir /opt/app-root/src/go/src/github.com/knative/client/windows
+ mv kn-workflow-windows-amd64.exe /opt/app-root/src/go/src/github.com/knative/client/windows/kn.exe
+ cp LICENSE /opt/app-root/src/go/src/github.com/knative/client/windows/
+ zip --quiet --junk-path - '/opt/app-root/src/go/src/github.com/knative/client/windows/*'

zip error: Nothing to do! (-)
Error: building at STEP "RUN sh -x /tmp/scripts/com.redhat.osl.cli.packager/install.sh": while running runtime: exit status 12
2024-06-03 18:55:32,012 tools.py:359        ERROR Command '['/usr/bin/podman', 'build', '--squash', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:1.33.0', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:latest', '--build-arg=KN_ARTIFACT_URL=http://download.devel.redhat.com/staging-cds/etera/test/kn-workflow-plugin/1/1.33/1.33.0-1/signed/', 'target/image']' returned non-zero exit status 12. Command stdout is 'None' with stderr 'None'
2024-06-03 18:55:32,012 cli.py:542        ERROR Could not run podman ['/usr/bin/podman', 'build', '--squash', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:1.33.0', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:latest', '--build-arg=KN_ARTIFACT_URL=http://download.devel.redhat.com/staging-cds/etera/test/kn-workflow-plugin/1/1.33/1.33.0-1/signed/', 'target/image']
Traceback (most recent call last):
  File "/home/rguimara/cekit-4.11.0/lib64/python3.11/site-packages/cekit/tools.py", line 351, in run_wrapper
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/podman', 'build', '--squash', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:1.33.0', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:latest', '--build-arg=KN_ARTIFACT_URL=http://download.devel.redhat.com/staging-cds/etera/test/kn-workflow-plugin/1/1.33/1.33.0-1/signed/', 'target/image']' returned non-zero exit status 12.
```

Comment (ricardozanini) [inline images/modules/com.redhat.osl.cli.packager/install.sh]:
```suggestion
mkdir "${KN_WORK_DIR}/windows" && mv kn-workflow-windows-amd64.exe "${KN_WORK_DIR}/windows/kn.exe" && cp LICENSE "${KN_WORK_DIR}/windows/" && zip --quiet --junk-path "${KN_WORK_DIR}/windows/*" > kn-workflow-windows-amd64.zip
```

Thy this

Comment (rgdoliveira) [reply images/modules/com.redhat.osl.cli.packager/install.sh]: With your suggestion the error changed to:

```
+ mkdir /opt/app-root/src/go/src/github.com/knative/client/windows
+ mv kn-workflow-windows-amd64.exe /opt/app-root/src/go/src/github.com/knative/client/windows/kn.exe
+ cp LICENSE /opt/app-root/src/go/src/github.com/knative/client/windows/
+ zip --quiet --junk-path '/opt/app-root/src/go/src/github.com/knative/client/windows/*'
Error: building at STEP "RUN sh -x /tmp/scripts/com.redhat.osl.cli.packager/install.sh": while running runtime: exit status 12
2024-06-03 19:39:43,281 tools.py:359        ERROR Command '['/usr/bin/podman', 'build', '--squash', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:1.33.0', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:latest', '--build-arg=KN_ARTIFACT_URL=http://download.devel.redhat.com/staging-cds/etera/test/kn-workflow-plugin/1/1.33/1.33.0-1/signed/', 'target/image']' returned non-zero exit status 12. Command stdout is 'None' with stderr 'None'
2024-06-03 19:39:43,281 cli.py:542        ERROR Could not run podman ['/usr/bin/podman', 'build', '--squash', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:1.33.0', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:latest', '--build-arg=KN_ARTIFACT_URL=http://download.devel.redhat.com/staging-cds/etera/test/kn-workflow-plugin/1/1.33/1.33.0-1/signed/', 'target/image']
Traceback (most recent call last):
  File "/home/rguimara/cekit-4.11.0/lib64/python3.11/site-packages/cekit/tools.py", line 351, in run_wrapper
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/podman', 'build', '--squash', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:1.33.0', '-t', 'openshift-serverless-1/logic-kn-workflow-cli-artifacts-rhel8:latest', '--build-arg=KN_ARTIFACT_URL=http://download.devel.redhat.com/staging-cds/etera/test/kn-workflow-plugin/1/1.33/1.33.0-1/signed/', 'target/image']' returned non-zero exit status 12.
```

Comment (ricardozanini) [reply images/modules/com.redhat.osl.cli.packager/install.sh]: This is:

```
12 /* nothing to do */ 13 /* missing or empty zip file */
```
Is this path correct?

Comment (mareknovotny) [reply images/modules/com.redhat.osl.cli.packager/install.sh]: shouldn't be there instead of `--junk-path` `--junk-paths` ?

Comment (rgdoliveira) [reply images/modules/com.redhat.osl.cli.packager/install.sh]: @mareknovotny this is a good point.

I fixed the local build using zip with `zip -jrq kn-workflow-windows-amd64.zip "${KN_WORK_DIR}/windows`  (I pushed a new commit)

I will try the build on OSBS too

**Review by ricardozanini — APPROVED** [review]

**Comment by rgdoliveira** [comment]
thanks, merging it.

**Pull request merged**
